### PR TITLE
fix: Create common type of `Location` in nav package

### PR DESCRIPTION
**Pull request by CodeBear801**
# Issue
https://github.com/Telenav/osrm-backend/issues/211

- Create common type of `Location` in nav package
- Remove redundant definitions in `Oasis`
- To do: 
   + Shall we make Location as Json object
   + Add serialize function and deserialize function
   + Add toString/fromString/toStringReverseLatLon/fromStringReverseLatLon

## Tasklist

 - [ ] review

### CODE REVIEW

File: integration/nav/doc.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/*
+package nav abstract navigation map related inforamation, such as
+- Location, latitude and longitude for specific point
+*/
+package nav
```
Comment (wangyoucao577) [inline integration/nav/doc.go]: Nit: how about `map`? E.g, `map.Location` vs. `nav.Location`.

**Comment by wangyoucao577** [comment]
> # Issue
> #211
> 
> * Create common type of `Location` in nav package
> * Remove redundant definitions in `Oasis`
> * To do:
>   
>   * Shall we make Location as Json object
>   * Add serialize function and deserialize function
>   * Add toString/fromString/toStringReverseLatLon/fromStringReverseLatLon
> 
> ## Tasklist
> * [ ]  review

Yes, I think it should provide a lot of ways for serialization, currently both json and string are useful.

File: integration/nav/doc.go (hunk 1 above)
Comment (wangyoucao577) [reply integration/nav/doc.go]: And how about move it into `integration/pkg/api/` folder?

[the remark this answers is not in the document]
Comment (CodeBear801) [reply integration/nav/doc.go]: I wrote this yesterday but didn't summit review......

Comment (CodeBear801) [reply integration/nav/doc.go]: Yes, that is the place this package should be.

Comment (CodeBear801) [reply integration/nav/doc.go]: > Nit: how about `map`? E.g, `map.Location` vs. `nav.Location`.

image

**Pull request merged**